portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1444,6 +1449,69 @@ protected CloseableHttpClient getCloseableHttpClient(HttpHost proxyHost) {
 		return _closeableHttpClient;
 	}
 
+	protected int getMaxURLDepth(
+		String url, int curLength, int maxLength, int count) {
+
+		String ampersand = StringPool.AMPERSAND;
+		String equal = StringPool.EQUAL;
+		String question = StringPool.QUESTION;
+
+		for (int i = 0; i < count; i++) {
+			ampersand = URLCodec.encodeURL(ampersand);
+			equal = URLCodec.encodeURL(equal);
+			question = URLCodec.encodeURL(question);
+		}
+
+		url = StringUtil.replace(url, question, ampersand);
+
+		List<String> redirectParams = new ArrayList<>();
+
+		String[] params = StringUtil.split(url, ampersand);
```

Comment:
It's easier to replace all the question marks with ampersands and split on ampersands than it is to split on both